engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/videoSettings/RenderingModuleSettingScreen.java

```diff
@@ -76,6 +78,13 @@ public void postInit() {
         renderingModuleRegistry.updateRenderingModulesOrder(moduleEnvironment, subContext);
         orderedModuleRenderingInstances = renderingModuleRegistry.getOrderedRenderingModules();
 
+        if (orderedModuleRenderingInstances == null || orderedModuleRenderingInstances.isEmpty()) {
```

Comment:
make  transformation "`null` -> emptyList" in `org/terasology/engine/module/rendering/RenderingModuleRegistry.java:121` instead checking on null.

Reply:
Odd. The only reason I was getting null-warnings on that site was because this method is explicitly annotated as `@Nullable`

https://github.com/MovingBlocks/Terasology/blob/73b04de01c6d543a897cd54f12e7fa6dcfa092b7/engine/src/main/java/org/terasology/engine/module/rendering/RenderingModuleRegistry.java#L75-L76

but I don't think it is? that field has a default value, is only set in one other place that looks like it's non-nullable, and is private so I don't think anything else can null it.

Reply:
why would you ever return null if you could return an empty list? o.O